src/operator.rs

```diff
@@ -835,6 +856,21 @@ impl UnaryPreAggOperator for Limit {
                     Ok(None)
                 }
             }
+            Limit::Tail { ref mut queue, limit } => {
+                if queue.len() == *limit {
+                    queue.pop_front();
+                }
+                queue.push_back(rec);
+
+                 Ok(None)
+            }
+        }
+    }
+
+    fn drain(self: Box<Self>) -> Box<Iterator<Item=Record>> {
```

Comment:
- consuming `Self` makes sense although I'm not sure why it needs to be a `Box<Self>`. 
- Mostly academic (there's not major downside to using a box here) but I think you can avoid it by defining `type RecIter=Iterator<Item=Record>` on UnaryPreAggOperator, then having `drain` directly return that type.

Reply:
Oh right! I forgot about sized. A big causes other problems because the variants aren't real types you can name